lib/vagrant-libvirt/action/create_domain_volume.rb

```diff
@@ -25,15 +25,15 @@ def call(env)
           @name = "#{env[:domain_name]}.img"
 
           # Verify the volume doesn't exist already.
-          domain_volume = ProviderLibvirt::Util::Collection.find_matching(
-            env[:machine].provider.driver.connection.volumes.all, @name
-          )
-          raise Errors::DomainVolumeExists if domain_volume
+          domain_volume = env[:machine].provider.driver.connection.volumes.all(
+            name: @name
+          ).first
+          raise Errors::DomainVolumeExists if domain_volume.id
```

Comment:
does the volumes.all always return something, otherwise I'd imagine this could result in a nil access error if the list is empty?

Reply:
I've a feeling I've simply forgotten how stuff works here over time, because I can't see how a vagrant up would work on a new instance if this did an attribute access and domain_volume was nil

Reply:
@ccrebolder looks like this is benefiting from a quirk in that use of a filter will always return an empty volume if no match is found rather than no volume. When I use a similar call with pools and a non-existing name, it returns an empty list, so it's likely this will change in the future and I think it would be worth checking `domain_volume` is not nil before testing `domain_volume.id` to guard against when this gets fixed upstream

Reply:
@electrofelix that's a good call, it does stand out as something fog will want to normalize one day. Will get reacquainted and work on this today, thanks for taking the time to look at it.

Reply:
I've put up fog/fog-libvirt#75 to trigger a conversation on it as well

Reply:
thanks, I'm checking for both `domain_volume` (if empty list is returned) and `domain_volume.id` (current behaviour of returning a single empty volume) being `nil` now in 669e5d9dfb34b82c93832397bf1518f4f4cb6482